diff --git a/components/Page.astro b/components/Page.astro
index 0ab90d1bacb10bb27ed1351f693358f8f90214f5..c8da2952de434caa3c6975525de55ed442bedd91 100644
--- a/components/Page.astro
+++ b/components/Page.astro
@@ -41,7 +41,7 @@ const pagefindEnabled =
 	lang={Astro.props.lang}
 	dir={Astro.props.dir}
 	data-has-toc={Boolean(Astro.props.toc)}
-	data-has-sidebar={Astro.props.hasSidebar}
+	data-has-sidebar={Astro.props.hasSidebar ? true : undefined}
 	data-has-hero={Boolean(Astro.props.entry.data.hero)}
 	data-theme="dark"
 >
diff --git a/integrations/virtual-user-config.ts b/integrations/virtual-user-config.ts
index 52f60badd4af05379ce94d9da587b674d4811ea1..027fd822f45aa84b6ec68e222fdd87488eff823b 100644
--- a/integrations/virtual-user-config.ts
+++ b/integrations/virtual-user-config.ts
@@ -79,7 +79,7 @@ export function vitePluginStarlightUserConfig(
 			: 'export const logos = {};',
 		'virtual:starlight/collection-config': `let userCollections;
 			try {
-				userCollections = (await import(${resolveId('./content/config.ts', srcDir)})).collections;
+				userCollections = (await import(${resolveId('./content.config.ts', srcDir)})).collections;
 			} catch {}
 			export const collections = userCollections;`,
 		'virtual:starlight/plugin-translations': `export default ${JSON.stringify(pluginTranslations)}`,
diff --git a/routes/common.astro b/routes/common.astro
index eab058a9185e884910f70dbaaae8821547d1a178..6d31300bfc2d18d574c2b046ea08d4e0cb477e11 100644
--- a/routes/common.astro
+++ b/routes/common.astro
@@ -1,4 +1,6 @@
 ---
+//@ts-expect-error
+import { render } from 'astro:content';
 import { generateRouteData } from '../utils/route-data';
 import type { Route } from '../utils/routing';
 import Page from '../components/Page.astro';
@@ -9,7 +11,7 @@ export type Props = {
 
 const { route } = Astro.props;
 
-const { Content, headings } = await route.entry.render();
+const { Content, headings } = await render(route.entry);
 const routeData = generateRouteData({ props: { ...route, headings }, url: Astro.url });
 ---
 
diff --git a/utils/routing.ts b/utils/routing.ts
index b365fb5f8e0f51b865f97e6a82365485532cdeaa..b0fa05d27d60a0dcd09e11e2b6494dd9fc98e3c1 100644
--- a/utils/routing.ts
+++ b/utils/routing.ts
@@ -51,9 +51,9 @@ const docs: StarlightDocsEntry[] = (
 		// In production, filter out drafts.
 		return import.meta.env.MODE !== 'production' || data.draft === false;
 	})) ?? []
-).map(({ slug, ...entry }) => ({
+).map((entry) => ({
 	...entry,
-	slug: normalizeIndexSlug(slug),
+	slug: normalizeIndexSlug(entry.id),
 }));
 
 function getRoutes(): Route[] {
